lib/expand/join.go

```diff
@@ -656,12 +660,8 @@ func (p *Peer) send(e install.Event) {
 	select {
 	case p.EventsC <- e:
 	case <-p.Context.Done():
-		select {
-		case p.EventsC <- install.Event{Error: trace.ConnectionProblem(p.Context.Err(), "context is closing")}:
```

Comment:
I don't think this is necessary as all critical loops are already listening on context expiration/cancellation.